packages/amplify-provider-awscloudformation/src/graphql-transformer/amplify-graphql-resource-manager.ts

```diff
@@ -266,6 +271,41 @@ export class GraphQLResourceManager {
     }
   };
 
+  private tableRecreationManagement = (currentState: DiffableProject, nextState: DiffableProject) => {
+    this.getTablesBeingReplaced().forEach(tableMeta => {
+      const ddbResource = this.getStack(tableMeta.stackName, currentState);
+      this.dropTable(tableMeta.tableName, ddbResource);
+      // clear any other states created by GSI updates as dropping and recreating supercedes those changes
+      this.clearTemplateState(tableMeta.stackName);
```

Comment:
if we toggle gsi changes between this then we might not have to clear the template state?

Reply:
what do you mean by "toggle gsi changes"?

Reply:
ah referring to the comment on `gsiManagement`